tests/physical_models/crop/test_root_dynamics.py

```diff
@@ -147,53 +147,109 @@ def test_wofost_pp_with_root_dynamics(self):
                 )
 
 
-class TestDiffRootDynamicsTDWI:
-    def test_gradients_tdwi_rd_root_dynamics(self):
+# Define parameters and outputs
+PARAM_NAMES = ["TDWI", "RDI", "RRI", "RDMCR", "RDMSOL", "IAIRDU", "RDRRTB"]
+OUTPUT_NAMES = ["RD", "TWRT"]
+
+# Define parameter configurations (value, dtype)
+PARAM_DEFAULT_VAL = {
+    "TDWI": (0.2, torch.float32),
+    "RDI": (0.2, torch.float32),
+    "RRI": (0.2, torch.float32),
+    "RDMCR": (0.2, torch.float32),
+    "RDMSOL": (0.2, torch.float32),
+    "IAIRDU": (0.2, torch.float32),
+    "RDRRTB": ([0.0, 0.0, 1.5, 0.02], torch.float32),  # Table parameter: [x1, y1, x2, y2]
+}
+
+# Define which parameter-output pairs should have gradients
+# Format: {param_name: [list of outputs that should have gradients]}
+GRADIENT_MAPPING = {
+    "TDWI": ["TWRT"],  # e.g. TDWI affects TWRT so I put it here
+    "RDI": ["RD"],
+    "RRI": ["RD"],
+    "RDMCR": ["RD"],
+    "RDMSOL": ["RD"],
+    "RDRRTB": ["TWRT"],
```

Comment:
This list provides useful information for users when calculating gradients for example in optimization. Can we add a comments about this in the docstring of root_dynamics? thanks!